pkg/server/commands/listusers/list_users_rpc_test.go

```diff
@@ -2,301 +2,1245 @@ package listusers
+
+				tuple.NewTupleKey("folder:x", "viewer", "user:maria"),
+				tuple.NewTupleKey("user:maria", "blocked", "user:maria"),
+			},
+			expectedUsers: []string{"user:will"},
+		},
 	}
+	tests.runListUsersTestCases(t)
+}
 
-	for _, test := range tests {
+func TestListUsersWildcards(t *testing.T) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
+	tests := ListUsersTests{
+		{
+			name: "direct_relationship_wildcard",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "viewer",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type: "user",
+					},
+				},
+			},
+			model: `model
+			schema 1.1
+			type user
+			type document
+				relations
+					define viewer: [user:*]`,
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "viewer", "user:*"),
+				tuple.NewTupleKey("document:2", "viewer", "user:*"),
+			},
+			expectedUsers: []string{"user:*"},
+		},
+		{
+			name: "direct_relationship_wildcard_with_direct_relationships_also",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "viewer",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type: "user",
+					},
+				},
+			},
+			model: `model
+			schema 1.1
+			type user
+			type document
+				relations
+					define viewer: [user:*,user]`,
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "viewer", "user:*"),
+				tuple.NewTupleKey("document:1", "viewer", "user:will"),
+				tuple.NewTupleKey("document:2", "viewer", "user:maria"),
+			},
+			expectedUsers: []string{"user:*", "user:will"},
+		},
+		{
+			name: "multiple_possible_wildcards_user_granularity",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "viewer",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type: "user",
+					},
+				},
+			},
+			model: `model
+			schema 1.1
+			type user
+			type group
+			type document
+				relations
+					define viewer: [ group:*, user:*]`,
 
-		ds := memory.New()
-		defer ds.Close()
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "viewer", "user:*"),
+				tuple.NewTupleKey("document:1", "viewer", "group:*"),
+			},
+			expectedUsers: []string{"user:*"},
+		},
+		{
+			name: "wildcard_with_indirection",
+			req: &openfgav1.ListUsersRequest{
+				Object:   &openfgav1.Object{Type: "document", Id: "1"},
+				Relation: "viewer",
+				UserFilters: []*openfgav1.ListUsersFilter{
+					{
+						Type: "user",
+					},
+				},
+			},
+			model: `model
+			schema 1.1
+			type user
+			type group
+				relations
+					define member: [user:*]
+			type document
+				relations
+					define viewer: [group#member]`,
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("document:1", "viewer", "group:eng#member"),
+				tuple.NewTupleKey("group:eng", "member", "user:*"),
+			},
+			expectedUsers: []string{"user:*"},
+		},
+	}
```

Comment:
should we have wildcard test for computed wildcard?   TTU?

Reply:
Good idea. Just added!